pkg/ssh/ssh.go

```diff
@@ -383,15 +386,18 @@ func (l *SSHTunnelList) pickTunnel(addr string) (tunnel, error) {
 		return nil, fmt.Errorf("No SSH tunnels currently open. Were the targets able to accept an ssh-key for user %q?", l.user)
 	}
 	// Prefer same tunnel as kubelet
-	// TODO: Change l.entries to a map of address->tunnel
-	for _, entry := range l.entries {
-		if entry.Address == addr {
-			return entry.Tunnel, nil
-		}
+	if entry, ok := l.entries[addr]; ok {
+		return entry.Tunnel, nil
 	}
 	klog.Warningf("SSH tunnel not found for address %q, picking random node", addr)
 	n := mathrand.Intn(len(l.entries))
-	return l.entries[n].Tunnel, nil
+	for address := range l.entries {
+		if n == 0 {
+			return l.entries[address].Tunnel, nil
+		}
+		n--
+	}
+	return nil, fmt.Errorf("Failed to pick a SSH tunnel")
```

Comment:
Make a note that this should never be reached. Also:

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
thanks. Addressed.